src/org/sosy_lab/java_smt/basicimpl/AbstractProver.java

```diff
@@ -158,4 +167,17 @@ public void close() {
     closeAllEvaluators();
     closed = true;
   }
+
+  @Override
+  public ShutdownManager getShutdownManagerForProver() throws UnsupportedOperationException {
+    return getShutdownManagerForProverImpl();
+  }
+
+  protected ShutdownManager getShutdownManagerForProverImpl() throws UnsupportedOperationException {
```

Comment:
As this method is not generic, we do not need it. just use inline it in the normal `getShutdoenManagerForProver`.
With a default implementation in the API, we can even remove this implmentation completely and just mark the overridden method as `abstract`.